compiler/rustc_hir_typeck/src/fn_ctxt/mod.rs

```diff
@@ -385,9 +385,35 @@ impl<'tcx> LoweredTy<'tcx> {
     }
 }
 
+fn never_type_behavior(tcx: TyCtxt<'_>) -> (DivergingFallbackBehavior, DivergingBlockBehavior) {
+    let (fallback, block) = parse_never_type_options_attr(tcx);
+    let fallback = fallback.unwrap_or_else(|| default_fallback(tcx));
+    let block = block.unwrap_or_default();
+
+    (fallback, block)
+}
+
+/// Returns the default fallback which is used when there is no explicit override via `#![never_type_options(...)]`.
+fn default_fallback(tcx: TyCtxt<'_>) -> DivergingFallbackBehavior {
+    use DivergingFallbackBehavior::*;
+
+    // Edition 2024: fallback to `!`
+    if tcx.sess.edition().at_least_rust_2024() {
+        return FallbackToNever;
+    }
+
+    // `feature(never_type_fallback)`: fallback to `!` or `()` trying to not break stuff
+    if tcx.features().never_type_fallback {
+        return FallbackToNiko;
```

Comment:
Do you have an idea what could be a better name? 

The reason I named it like that, is that it is inherently confusing, in a way. + it's probably on the way to be removed.

Reply:
`ContextDependent`

also, instead of naming the variants `FallbackToX`, I personally prefer using `DivergingFallbackBehavior::ToNever` and stop using a blob import for the variants

Reply:
`DivergingFallbackBehavior::ToNever ` is a bit too long and annoying IMO, but sure. `ContextDependent` sounds good.

Should I rename things in this PR?

Reply:
i'd like that, given that these are style nits I don't consider them to be blockin though